### ISSUE TITLE
gasUsed values from getTransactionReceipt should be in wei to avoid confusion and rounding issues

### ISSUE THREAD

**Issue by NatPDeveloper** [ethereum/web3.js#6360]
#### Expected behavior

gasUsed returned from await web3.eth.getTransactionReceipt(event.transactionHash); is in wei

#### Actual behavior

It's returned in gwei with an additional decimal place of precision

1911959n -> 191195.9 gwei

#### Steps to reproduce the behavior

1. Use the above function with a transaction receipt from arbitrum.

#### Environment

"web3": "^4.0.3", ubuntu 20, nodejs v16.18.0

---

If this can be detailed in the docs or done in wei would be great.

One would expect to get to wei you would to `const gasPaid = BigInt(receipt.gasUsed) * 10n ** 9n;`

But you must actually do `const gasPaid = BigInt(receipt.gasUsed) * 10n ** 8n;` which is not very intuitive.

**Comment by jdevcs** [comment]
Currently being in compliance with client, we return `gwei`. Have you considered [using](https://docs.web3js.org/api/web3-utils/function/toWei) 
`web3.utils.toWei(gasUsed, 'gwei')`

**Comment by NatPDeveloper** [comment]
> Currently being in compliance with client, we return `gwei`. Have you considered [using](https://docs.web3js.org/api/web3-utils/function/toWei) `web3.utils.toWei(gasUsed, 'gwei')`

Thank you for the response.

This yields improper precision since there is one additional decimal place of precision added.

```
web3.utils.toWei("1911959", 'gwei')
'1911959000000000'
> 1911959000000000 / 1e9
1911959 // should be 191195.9
```

**Comment by Muhammad-Altabba** [comment]
Hello @NatPDeveloper ,
First, let us agree that 1911959n is indeed 1911959. No decimal place here. And the `n` at the end of the number is from Javascript just to note that the number is a [BigInt](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt).
Second when you call `web3.utils.toWei("1911959", 'gwei')` the correct value is: '1911959000000000'. And it is the same number that is originally in `gwei` returned after adding 9 zeros to its right so it becomes in `wei`. You can check this exact behavior in: https://eth-converter.com/.

Does this resolve your concerns? Please, let us know. If not, please, check first with https://eth-converter.com/ and then provide again your findings if web3.js provides different values than https://eth-converter.com/.

Thanks,

**Comment by NatPDeveloper** [comment]
Perhaps I'm not being clear. Let me provide example which you can reproduce and see that what you're proposing does not return the value you expect.

<details>
<summary>Example</summary>

```
const Web3 = require("web3");
const web3 = new Web3("https://arb1.arbitrum.io/rpc")
web3.eth.getTransactionReceipt("0xb8ee30a61d946c6fb5f8340ad71dfa1580fb28e17918a92bc1c03a358c246cc6").then(receipt => console.log(receipt))
> {
  blockHash: '0x7bfd075264c6b2da4993b4da3567da9f935a0b0b8bd6f7e45c6c67cc56ffa4f5',
  blockNumber: 128330666,
  contractAddress: null,
  cumulativeGasUsed: 5469836,
  effectiveGasPrice: 100000000,
  from: '0x19e652baf0b69795d270ec3b52bcce80dffcfd7b',
  gasUsed: 706484,
  gasUsedForL1: '0x94b7e',
  l1BlockNumber: '0x113c180',
  logs: [
    {
      address: '0xD27d347A9968a2e393d2991BCC65d8c81755aAC1',
      topics: [Array],
      data: '0x',
      blockNumber: 128330666,
      transactionHash: '0xb8ee30a61d946c6fb5f8340ad71dfa1580fb28e17918a92bc1c03a358c246cc6',
      transactionIndex: 3,
      blockHash: '0x7bfd075264c6b2da4993b4da3567da9f935a0b0b8bd6f7e45c6c67cc56ffa4f5',
      logIndex: 8,
      removed: false,
      id: 'log_843b1047'
    }
  ],
  logsBloom: '0x00000000000000000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000008000000000000000000000000000000008000000000000000000000000000000000000000000000000000000000000000000020000080000000000000000000010000000000000000000000020000000000000080000000000040000000000000000000000000000000000000000000000001000000000000000000000000000000000000200000002000000000000000000020000000000000000001000000000000000000000000000000000010000000000000000000000000000000000000000000000',
  status: true,
  to: '0xd27d347a9968a2e393d2991bcc65d8c81755aac1',
  transactionHash: '0xb8ee30a61d946c6fb5f8340ad71dfa1580fb28e17918a92bc1c03a358c246cc6',
  transactionIndex: 3,
  type: '0x2'
}
```
</details>

Reported: `gasUsed: 706484`

```
> web3.utils.toWei("706484", 'gwei')
'706484000000000'
> 706484000000000 / 1e18
0.000706484
```

That value is one decimal place off. If you view the trx [here](https://arbiscan.io/tx/0xb8ee30a61d946c6fb5f8340ad71dfa1580fb28e17918a92bc1c03a358c246cc6). You will see

**Transaction Fee:**
0.0000706484 ETH ($0.12) // 4 leading 0's

**Versus what was calculated:**
0.000706484 // 3 leading 0's

So I recommend the value returned is in wei. The returned value is actually in gwei with one additional decimal place of precision (I realize this is fixed point, but if you bring it down to floating point like I did above, it's off). 

I'm assuming the extra place was added for additional precision but it really should just be returned in wei or actual gwei as the average developer has little hope of noticing this discrepency.

If it is not changed, please make a note in your docs.

---

To be even more clear, if you remove the last 4 (**706484 -> 70648**), then the amounts are the same, minus the additional precision.

```
> web3.utils.toWei("70648", 'gwei') // removed the last '4'
'70648000000000'
> 70648000000000 / 1e18
0.000070648 // 4 leading 0's
```

---

Sanity check

```
> 1_000_000_000_000_000_000 / 1e18
1
```

So we agree that `1e18` is 1 with 18 decimals behind it.

**Comment by Muhammad-Altabba** [comment]
Hi @NatPDeveloper,
Thanks for your illestration. I checked how Arbitrum gas estimate is done. And I can see from where the confusion came. 
As a ground info, Arbitrum follows a different aproche for gas calculation in which it depends on some concept called base gas fee. And this base gas fee can be read using [their sdk](https://github.com/OffchainLabs/arbitrum-sdk/blob/main/src/lib/utils/lib.ts#L12):
```ts
export const getBaseFee = async (provider: Provider) => {
  const baseFee = (await provider.getBlock('latest')).baseFeePerGas
  if (!baseFee) {
    throw new ArbSdkError(
      'Latest block did not contain base fee, ensure provider is connected to a network that supports EIP 1559.'
    )
  }
  return baseFee
}
```
Let me describe it to you using your own example:
> Reported: gasUsed: 706484

This reported gas is not in wei nor in gwei. So calling  `web3.utils.toWei("706484", 'gwei')` follows the principle garbage-in-garbage-out. And to  get the unit of this gasUsed you need to know the base gas fee which could be different for each block. And it is in 0.1 Gwei for the block in which the following transaction exist, for example: https://arbiscan.io/tx/0xb8ee30a61d946c6fb5f8340ad71dfa1580fb28e17918a92bc1c03a358c246cc6
![image](https://github.com/web3/web3.js/assets/24407834/56422881-793f-4a8c-8bb2-de136b163496)
So, in short you need to query the base gas fee to know the measurement unit.  And for your provided example, the 706484 is in 0.1 Gwei and not in 1 Gwei.
If you have further questions. Related to calculating gas in Arbitrum. I suggest you open an issue at their repository. And if you think that web3.js should handle this for Arbitrum, I encourage you to write [a plugin for web3.js](https://docs.web3js.org/guides/web3_plugin_guide/plugin_authors) that handle its specifications. 
All the best,
